### PR TITLE
Added photo download flag

### PR DESCRIPTION
**Pull request by DouglasKrouth**
* Added flag to download photos; not clever, just downloads them in current running dir. Easy feature request would be to have a default/specified dir that's used to store images.
* Added use of logger instead of `print()` statements
* Docstrings

The script is reaching a threshold where it's passing around a lot args for internal state. Function signatures would be cleaner if we bundled them as class methods.

[Original Issue](https://github.com/bellingcat/telegram-phone-number-checker/issues/31)

### CODE REVIEW

File: telegram_phone_number_checker/main.py [hunk 1]
```diff
@@ -80,6 +90,32 @@ async def get_names(client: TelegramClient, phone_number: str) -> dict:
                     "phone": user.phone,
                 }
             )
+        if download_profile_photos is True:
+            try:
+                photo_output_path = Path("{}_{}_photo.jpeg".format(user.id, user.phone))
```
Comment (GalenReich) [inline telegram_phone_number_checker/main.py:95]: It looks like `user.phone` is often `None`, let's use the user supplied `phone_number` instead

File: telegram_phone_number_checker/main.py [hunk 2]
```diff
@@ -80,6 +90,32 @@ async def get_names(client: TelegramClient, phone_number: str) -> dict:
                     "phone": user.phone,
                 }
             )
+        if download_profile_photos is True:
```
Comment (GalenReich) [inline telegram_phone_number_checker/main.py:93]: This block should be indented by one more level so it lives inside the `elif number_of_matches == 1:` block and doesn't interfere with the final `else`

File: telegram_phone_number_checker/main.py (hunk 1 above)
Comment (GalenReich) [inline telegram_phone_number_checker/main.py:95]: It looks like `user.phone` is often `None`, let's use the user-supplied `phone_number` instead

**Review by GalenReich — CHANGES_REQUESTED** [review]
In general, this looks great, just a couple of minor tweaks - thanks for implementing logging too, that's a nice enhancement 🚀

File: telegram_phone_number_checker/main.py (hunk 1 above)
Comment (GalenReich) [reply telegram_phone_number_checker/main.py:95]: (there are a couple of other instances in the block, not just here)

File: telegram_phone_number_checker/main.py (hunk 2 above)
Comment (DouglasKrouth) [reply telegram_phone_number_checker/main.py]: Fixed here f049059c2f0d8b79a844fb6f9253e0169bd66f34

File: telegram_phone_number_checker/main.py (hunk 1 above)
Comment (DouglasKrouth) [reply telegram_phone_number_checker/main.py]: f049059c2f0d8b79a844fb6f9253e0169bd66f34

**Review by GalenReich — APPROVED** [review]
This is great, thank you for taking the time to write this and make those changes! Please feel free to take a look at any other projects if you are interested, and if you'd like to get involved with our wider community, we have a [Discord](https://discord.gg/bellingcat) too - thanks again! 🙌

**Pull request merged**